packages/auth-providers/netlify/setup/src/templates/api/lib/auth.ts.template

```diff
@@ -58,62 +56,47 @@ export const isAuthenticated = (): boolean => {
 
 /**
  * When checking role membership, roles can be a single value, a list, or none.
- * You can use Prisma enums too (if you're using them for roles), just import your enum type from `@prisma/client`
+ * You can use Prisma enums too (if you're using them for roles), just import
+ * your enum type from `@prisma/client`
  */
 type AllowedRoles = string | string[] | undefined
 
 /**
- * Checks if the currentUser is authenticated (and assigned one of the given roles)
+ * Checks if the currentUser is authenticated (and assigned one of the given
+ * roles)
  *
- * @param roles: {@link AllowedRoles} - Checks if the currentUser is assigned one of these roles
+ * @param roles: {@link AllowedRoles} - Checks if the currentUser is assigned
+ *   one of these roles
  *
- * @returns {boolean} - Returns true if the currentUser is logged in and assigned one of the given roles,
- * or when no roles are provided to check against. Otherwise returns false.
+ * @returns {boolean} - Returns true if the currentUser is logged in and
+ *   assigned one of the given roles, or when no roles are provided to check
+ *   against. Otherwise returns false.
  */
-export const hasRole = (roles: AllowedRoles): boolean => {
-  if (!isAuthenticated()) {
+export const hasRole = (allowedRoles: AllowedRoles): boolean => {
+  if (!isAuthenticated() || !allowedRoles) {
     return false
   }
 
-  const currentUserRoles = context.currentUser?.roles
-
-  if (typeof roles === 'string') {
-    if (typeof currentUserRoles === 'string') {
-      // roles to check is a string, currentUser.roles is a string
-      return currentUserRoles === roles
-    } else if (Array.isArray(currentUserRoles)) {
-      // roles to check is a string, currentUser.roles is an array
-      return currentUserRoles?.some((allowedRole) => roles === allowedRole)
-    }
-  }
-
-  if (Array.isArray(roles)) {
-    if (Array.isArray(currentUserRoles)) {
-      // roles to check is an array, currentUser.roles is an array
-      return currentUserRoles?.some((allowedRole) =>
-        roles.includes(allowedRole)
-      )
-    } else if (typeof currentUserRoles === 'string') {
-      // roles to check is an array, currentUser.roles is a string
-      return roles.some((allowedRole) => currentUserRoles === allowedRole)
-    }
-  }
+  const currentUserRoles = context.currentUser.roles
+  const roles = typeof allowedRoles === 'string' ? [allowedRoles] : allowedRoles
 
-  // roles not found
-  return false
+  return currentUserRoles.some((role) => roles.includes(role))
```

Comment:
@jtoar Can you please just give this a quick look to make sure I've not simplified it too much.

Reply:
It looks good to me, especially since you set a default for roles to `[]`. The original that we've had here for a while was pretty tedious to read so glad you're simplifying it!